cfgov/v1/atomic_elements/atoms.py

```diff
@@ -112,6 +112,33 @@ class Button(Hyperlink):
     ], default='regular')
 
 
+class ImageBasicStructValue(blocks.StructValue):
+    @property
+    def url(self):
+        upload = self.get('upload')
+
+        if upload:
+            return upload.get_rendition('original').url
+
+    @property
+    def alt_text(self):
+        # TODO: This duplicates the logic in v1.jinja2tags.image_alt_value,
+        # which cannot be called here because of a circular import. It would
+        # be better to deprecate the image_alt_value tag in favor of using
+        # this logic wherever we use ImageBasic atoms.
+        alt = self.get('alt')
+        if alt:
+            return alt
+
+        upload = self.get('upload')
+        if upload:
```

Comment:
Technically this isn't necessary because [`alt` can't be null](https://github.com/cfpb/cfgov-refresh/blob/7c95ee016c20480a6085031ce9685fb063309cb8/cfgov/v1/models/images.py#L13). But I agree this will add some clarity to the code, so I'll add it.

Reply:
NOt sure if this is a null vs. blank vs. none thing that I still can't seem to wrap my head around, but the `alt` field definitely does not have to be _populated,_ and if it's not populated, then we have no reason to return it.

Reply:
We want to return it because it should be the one place that defines the logic for what an image's alt text is. We're not just returning the value, we're saying that we defer the logic for the alt text to that method. And, because of the way CFGOVImage is defined, that can never be null and will always default to the empty string.

Actually thinking about this some more, and your other question [here](https://github.com/cfpb/cfgov-refresh/pull/5781#discussion_r442510301), I think I'd prefer to back this out to

```py
alt = self.get('alt')
if alt:
    return alt

upload = self.get('upload')
if upload:
    return upload.alt

return None
```

(See latest commit.)

The logic should be:

1. If the block has an `alt` defined, return it.
2. Otherwise, return the image's `alt` <-- this defers to the image logic about handling alt text, which encapsulates falling back to the empty string.
3. In the case where `upload` is undefined, its alt text should be undefined. We shouldn't  ever be putting a block like this into a template.